share/eds/store.go

```diff
@@ -0,0 +1,207 @@
+package eds
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/filecoin-project/dagstore"
+	"github.com/filecoin-project/dagstore/index"
+	"github.com/filecoin-project/dagstore/mount"
+	"github.com/filecoin-project/dagstore/shard"
+	"github.com/ipfs/go-datastore"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+
+	"github.com/celestiaorg/celestia-node/share"
+
+	"github.com/celestiaorg/rsmt2d"
+)
+
+const (
+	blocksPath     = "/blocks/"
+	indexPath      = "/index/"
+	transientsPath = "/transients/"
+)
+
+type EDSStore struct { //nolint:revive
+	dgstr  *dagstore.DAGStore
+	bs     blockstore.Blockstore
+	mounts *mount.Registry
+
+	topIdx index.Inverted
+	carIdx index.FullIndexRepo
+
+	basepath string
+}
+
+func NewEDSStore(basepath string, ds datastore.Batching) (*EDSStore, error) {
+	err := setupPath(basepath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to setup EDSStore directories: %w", err)
+	}
+
+	r := mount.NewRegistry()
+	err = r.Register("fs", &mount.FSMount{FS: os.DirFS(basepath + blocksPath)})
+	if err != nil {
+		return nil, fmt.Errorf("failed to register FS mount on the registry: %w", err)
+	}
+
+	fsRepo, err := index.NewFSRepo(basepath + indexPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create index repository: %w", err)
+	}
+
+	invertedRepo := index.NewInverted(ds)
+	dagStore, err := dagstore.NewDAGStore(
+		dagstore.Config{
+			TransientsDir: basepath + transientsPath,
```

Comment:
I have to look into it deeper. It would make sense that we don't use them according to the description of the transients directory, but we definitely create transients, and unused transients are actually what get cleaned up during GC.

Reply:
AFAIU, transients are created during fetch of remote mounts

Reply:
```
func TestTransientExists(t *testing.T) {
	ctx, cancel := context.WithCancel(context.Background())
	t.Cleanup(cancel)

	edsStore, err := newStore(t)
	require.NoError(t, err)
	err = edsStore.Start(ctx)
	require.NoError(t, err)

	eds, dah := randomEDS(t)
	err = edsStore.Put(ctx, dah, eds)
	assert.NoError(t, err)

	files, err := os.ReadDir(edsStore.basepath + transientsPath)
	fmt.Println(files[0].Name())
}
```

Prints:
```
=== RUN   TestTransientExists
transient-13F9FB2771E0545486363633AE523764EEDE9BE96102AA9BE3733F643D814AC6.complete
--- PASS: TestTransientExists (0.01s)
PASS
```

Reply:
I see